### ISSUE TITLE
can't recognize the right type for the return value of np.dot

### ISSUE THREAD

**Issue by Cydiater** [microsoft/pyright#1402]
**Describe the bug**
Pyright seemed not be able to recognize the return value of np.dot

**To Reproduce**
try run the following code
```python
import numpy as np

def hack(a, b):
    c = np.dot(a, b)
    np.max(c, axis = 1, keepdims = True)
```
**Expected behavior**
```
No configuration file found.
stubPath /Users/cydiater/demo/typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
/Users/cydiater/demo/demo.py
  5:5 - error: No overloads for "np.max(c, axis=1, keepdims=True)" match parameters
    Argument types: (bool_, Literal[1], Literal[True]) (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
Completed in 0.684sec
```

**VS Code extension or command-line**
command-line

**Comment by erictraut** [comment]
Thanks for the bug report. The TypeVar constraint solving logic was not properly handling the case where an "Any" or "Unknown" was assigned to a constrained TypeVar. (The np.dot function parameters are annotated with `_Int`, which is a constrained TypeVar.)

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.104, which I just published. It will also be included in the next release of pylance.
